task-list/app/views/tasks/_form.html.erb

```diff
@@ -0,0 +1,13 @@
+<%= form_for @task, url: {action: "#{@action}"} do |f| %>
```

Comment:
Nice use of a partial that changes the behavior of the form based on variables from the controller!